src/controllers/authController.js

```diff
@@ -85,13 +94,13 @@ const loginUser = async (req, res) => {
     }
     const token = generateToken(user._id);
 
-    const userObj = user.toObject();
-    delete userObj.password;
+    const userInfo = user.toObject();
+    delete userInfo.password;
 
     res.status(StatusCodes.OK).json({
```

Comment:
Are we using http only cookies ?

res.cookie('token', token, {
  httpOnly: true,
  secure: process.env.NODE_ENV === 'production', // Ensure the cookie is sent over HTTPS in production
  expires: new Date(Date.now() + 3600000), // 1 hour
});
res.status(StatusCodes.OK).json({
  message: 'Login successful',
});

Should happen for login and register, instead of sending it over json.